### PR TITLE
Add content for Nodes, Workloads, Dashboard and Security Events and JIRA NVSHAS-7522, NVSHAS-9293 and NVSHAS-8148

### PR DESCRIPTION
**Pull request by xingzhang-suse**

### CODE REVIEW

**Comment by nunix** [comment]
@xingzhang-suse can I kindly ask you to add the `.idea` to your `.gitignore` file that should be located at the root of the repo.

**Comment by xingzhang-suse** [comment]
> @xingzhang-suse can I kindly ask you to add the `.idea` to your `.gitignore` file that should be located at the root of the repo.

Sure, I have removed .idea folder from code and add it into .gitignore.

File: docs/06.scanning/01.scanning/01.vulnerabilities/01.vulnerabilities.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ Use the filter box to enter a string match, or use the advanced filter next to i
 
 ![AdvancedFilter](advanced_filter_4.png)
 
-Selecting any CVE listed provides additional details about the CVE, remediation, and which images, nodes, or containers are Impacted. The Protection State icon (circle) shows various colors to indicate a rough percentage of the impacted items which are unprotected by NeuVector during run-time, protected by NeuVector rules (in a Monitor or Protect mode), or unaffected in run-time (e.g. an image scanned with this vulnerability has no running containers). The Protection State column color scheme is:
+Selecting any CVE listed provides additional details about the CVE, remediation, and which images, nodes, or containers are Impacted. The Protection State icon (circle) shows various colors to indicate a rough percentage of the impacted items which are unprotected by NeuVector during run-time, protected by NeuVector rules (in a Monitor or Protect mode), or unaffected in run-time (e.g. an image scanned with this vulnerability has no running containers). Images include registry's static images and run-time container's images. Run-time container's images have the same colors as their containers. The Protection State column color scheme is:
```
Comment (sunilarjun) [inline docs/06.scanning/01.scanning/01.vulnerabilities/01.vulnerabilities.md:40]:
```suggestion
Selecting any CVE listed provides additional details about the CVE, remediation, and which images, nodes, or containers are Impacted. The Protection State icon (circle) shows various colors to indicate a rough percentage of the impacted items which are unprotected by NeuVector during run-time, protected by NeuVector rules (in a Monitor or Protect mode), or unaffected in run-time (e.g. an image scanned with this vulnerability has no running containers). Images include the registry's static images and the run-time container's images. The run-time container's images have the same colors as their containers. The Protection State column color scheme is:
```

File: docs/06.scanning/02.registry/02.registry.md [hunk 2]
```diff
@@ -27,6 +27,14 @@ The registry will be scanned according to a schedule, which is configurable. By
 
 After the scan is completed you will see the results below it. Click on the repository/tag to see vulnerabilities and click on the vulnerability to see more info. You can also download the report in a CSV file or see the results in the Event logs.
 
+From v5.4.0 NeuVector provide "View all scanned images". Users don't have to look into registry to see a specific image. Instead, users can click the last row to get all scanned image list in one stop.
```
Comment (sunilarjun) [inline docs/06.scanning/02.registry/02.registry.md:30]:
```suggestion
From v5.4.0 NeuVector provides a "View all scanned images" option. Users don't have to look into a registry to see a specific image. Instead, users can click the last row to list all scanned images.
```

File: docs/06.scanning/02.registry/02.registry.md [hunk 3]
```diff
@@ -27,6 +27,14 @@ The registry will be scanned according to a schedule, which is configurable. By
...
+
+![Registry](view-all-scanned-images.png)
+
+Users can also land to Registries page and open specific image from the hyperlink from Containers page. If the image name is not clickable, it means that the image has not been scanned yet.
```
Comment (sunilarjun) [inline docs/06.scanning/02.registry/02.registry.md:34]:
```suggestion
Users can also land on the Registries page and open specific images from the hyperlink from the Containers page. If the image name is not clickable, it means that the image has not been scanned yet.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+title: 5.x Dashboard
+taxonomy:
+    category: docs
+slug: /ui_extension/dashboard
+---
+
+
+### Monitor overview of security status
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:9]:
```suggestion
### Monitor Overview of Security Status
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+In UI Extension on Rancher portal, NeuVector's Dashboard works similar as its standalone page. It provides security risk score and factors to let user know the security status of the clusters. In the bottom panel, it provides 4 tabs to let user switch the charts to monitor different categories.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:11]:
```suggestion
In the UI Extension in the Rancher portal, NeuVector's Dashboard works similar as its standalone page. It provides security risk score factors to let users know the security status of their clusters. In the bottom panel, it provides 4 tabs to let users switch the charts to monitor different categories.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+In Security Events tab, the line chart shows a date-based event count in a timeline.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:13]:
```suggestion
In the Security Events tab, the line chart shows a date-based event count in a timeline.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+The top security events bar charts presents the top 5 source and destination endpoints.
+
+![Dashboard](dashboard_security_events.png)
+
+In Ingress and Egress Exposure tab, the stacked bar chart shows amount of exposed endpoints by actions.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:18]:
```suggestion
In Ingress and Egress Exposure tab, the stacked bar chart shows the amount of exposed endpoints by actions.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+On the right side, the exposure tables show egress and ingress endpoint service with scan informations.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:19]:
```suggestion
On the right side, the exposure tables show egress and ingress endpoint service with scan information.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+![Dashboard](dashboard_exposure.png)
+
+Click on the row of egress/ingress table, the table shows conversations on specific service. The FQDN name / IP address is shown with geography location based national flag in External Host field.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:23]:
```suggestion
Click on the row of egress/ingress table, the table shows conversations on specific services. The FQDN name / IP address is shown with a geography location based national flag in the External Host field.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+![Dashboard](dashboard_conversations.png)
+
+Click on the External Host, it redirects to whois.com to show the IP address' detail informations.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:27]:
```suggestion
Click on the External Host, it redirects to whois.com to show the IP address' detail information.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+![External](whois.png)
+
+In Policy Mode tab, the pie charts shows the ratio and numbers of pods / services under different policy mode.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:31]:
```suggestion
In the Policy Mode tab, the pie charts show the ratio and numbers of pods / services under different policy modes.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+![Dashboard](dashboard_policy_mode.png)
+
+In Top Vulnerable Assets tab, the bar charts shows the top 5 pods / nodes which contain high or medium severity vulnerabilities.
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:35]:
```suggestion
In the Top Vulnerable Assets tab, the bar charts show the top 5 pods / nodes which contain high or medium severity vulnerabilities.
```

File: docs/15.ui_extension/01.dashboard/01.dashboard.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+![Dashboard](dashboard_top_vul_assets.png)
+
+### Not included functions comparing with NeuVector standalone page
```
Comment (sunilarjun) [inline docs/15.ui_extension/01.dashboard/01.dashboard.md:39]:
```suggestion
### Functions Not Included When Comparing with NeuVector Standalone Page
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+title: 5.x Runtime Security
+taxonomy:
+category: docs
+slug: /ui_extension/runtime_security
+---
+
+
+### Assets Security Scan Result
+
+As Rancher has its own Nodes and Workloads page to manage and monitor the assets' status, NeuVector injects components to display its runtime scan result.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:11]:
```suggestion
As Rancher has its own Nodes and Workloads page to manage and monitor the assets statuses, NeuVector injects components to display its runtime scan result.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+
+#### Nodes
+
+In nodes summary table, vulnerabilities column is injected to show each node's detected vulnerabilities amount by high / medium severity.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:15]:
```suggestion
In the nodes summary table, the vulnerabilities column is injected to show each node's detected vulnerabilities amount by high / medium severity.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+![Nodes](nodes.png)
+
+Click on the vulnerabilities cell, a slide panel comes from right side of the page to show scan result details. The database version of the scanner is shown as well as a filterable vulnerability list.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:19]:
```suggestion
Click on the vulnerabilities cell, a slide panel will come from the right side of the page to show scan result details. The database version of the scanner is shown as well as a filterable vulnerability list.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+![Nodes](nodes_vul.png)
+
+Click on the node name cell on nodes table, it redirects to node detail page. NeuVector inject a tab to show vulnerability list.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:23]:
```suggestion
Click on the node name cell on nodes table, it redirects to the node detail page. NeuVector injects a tab to show the vulnerability list.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+
+![Nodes](node_detail_vul.png)
+
+Click on the CVE name cell on the vulnerabilities table, it opens a popup to show CVE description with an external link on the header to SUSE's CVE information website.
+
+![Nodes](node_detail_vul_cve.png)
+
+![SUSE](suse_cve.png)
+
+A CSV file downloading button is provided above the vulnerabilities table. The file data includes all the fields which are in the table.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:33]:
```suggestion
A CSV file download button is provided above the vulnerabilities table. The file data includes all the fields which are in the table.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+
+![CSV](download_vul.png)
+
+![CSV](cve_csv.png)
+
+#### Workloads
+
+Temporarily, vulnerabilities column will not be injected in workloads table in v5.4 release due to potential performance concern. NeuVector team will work with Rancher team to deal with it and integrate it soon.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:41]:
```suggestion
Temporarily, the vulnerabilities column will not be injected in the workloads table in the v5.4 release due to potential performance concern. The NeuVector team will work with the Rancher team to integrate it soon.
```

File: docs/15.ui_extension/02.runtime _security/runtime_security.md [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+![Workload](pods.png)
+
+Click on a pod name on the workloads table, it redirects to pod detail page. NeuVector inject a tab to show vulnerability list.
```
Comment (sunilarjun) [inline docs/15.ui_extension/02.runtime _security/runtime_security.md:45]:
```suggestion
Click on a pod name in the workloads table, it redirects to the pod detail page. NeuVector injects a tab to show the vulnerability list.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+category: docs
+slug: /ui_extension/notifications
+---
+
+
+### Event Reporting
+
+#### Security Events
+
+In UI Extension on Rancher portal, Security Events works similar as NeuVector standalone page.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:13]:
```suggestion
In the UI Extension on the Rancher portal, Security Events work similar to the NeuVector standalone page.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+The events can be triggered by several factors
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:14]:
```suggestion
The events can be triggered by several factors:
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+
+- Dangerous conversation or data transportation from somewhere to the endpoint (Node or pod)
+- End-to-end violated connection against network policy
+- Violated operation against process profile rule
+- Violated operation against file access rule
+
+![Security Events](sec_events.png)
+
+Slide left or  / and right dot on the time slider to filter events by date.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:23]:
```suggestion
Slide the dot left or right on the time slider to filter events by date.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](sec_events_time_filter.png)
+
+Type in keyword into quick filter box to filter any event which includes the keyword.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:27]:
```suggestion
Type in keywords into the quick filter box to filter any event which includes the keyword.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](sec_events_quick_filter.png)
+
+Open the Advanced filter slide panel, there are multiple options supporting your searching.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:31]:
```suggestion
Open the Advanced filter slide panel, and you will find multiple options supporting your search:
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+
+- Severity, Location and Category can be filtered according to the tags on the right side of the event title.
+- Autocomplete list can support user to enter Host, Source, Destination
+- Autocomplete list also support multiple tag-input box for Namespace
+
+![Security Events](sec_events_adv_filter.png)
+
+To read the details of the event, click on a record to expand the panel. In the message box, it contains more informations.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:39]:
```suggestion
To read the details of the event, click on a record to expand the panel. In the message box you can view more information.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 27]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](sec_event_detail.png)
+
+Host name is clickable for opening host detail popup. Vulnerabilities and Group have external link to redirect to NeuVector SSO page.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:43]:
```suggestion
The Host name is clickable for opening the host detail popup. Vulnerabilities and Group have an external link to redirect to the NeuVector SSO page.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](node_detail_open.png)
+
+Workload name is clickable for opening workload detail popup. Group has external link to redirect to NeuVector SSO page.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:47]:
```suggestion
The Workload name is clickable for opening the workload detail popup. Group has an external link to redirect to the NeuVector SSO page.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 29]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](pod_detail_open.png)
+
+Reported by field in the message box is clickable for opening enforcer detail popup.
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:51]:
```suggestion
The Reported by field in the message box is clickable and opens the enforcer detail popup.
```

File: docs/15.ui_extension/03.notifictions/notifications.md [hunk 30]
```diff
@@ -0,0 +1,59 @@
...
+
+![Security Events](enforcer_detail_open.png)
+
+### Not included functions comparing with NeuVector standalone page
```
Comment (sunilarjun) [inline docs/15.ui_extension/03.notifictions/notifications.md:55]:
```suggestion
### Functions Not Included When Comparing with NeuVector Standalone Page
```

**Review by sunilarjun — CHANGES_REQUESTED** [review]
Hello! I've made some edits with adjusting grammar, once fixed this PR should be good to merge in. Thank you!

File: docs/06.scanning/01.scanning/01.vulnerabilities/01.vulnerabilities.md (hunk 1 above)
Comment (xingzhang-suse) [reply docs/06.scanning/01.scanning/01.vulnerabilities/01.vulnerabilities.md]: Fixed

File: docs/06.scanning/02.registry/02.registry.md (hunk 3 above)
Comment (xingzhang-suse) [reply docs/06.scanning/02.registry/02.registry.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 4 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 5 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 6 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 7 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 8 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 9 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 10 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 11 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 12 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/01.dashboard/01.dashboard.md (hunk 13 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/01.dashboard/01.dashboard.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 14 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 15 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 16 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 17 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 18 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 19 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/02.runtime _security/runtime_security.md (hunk 20 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/02.runtime _security/runtime_security.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 21 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 22 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 23 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 24 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 25 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 26 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 27 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 28 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 29 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

File: docs/15.ui_extension/03.notifictions/notifications.md (hunk 30 above)
Comment (xingzhang-suse) [reply docs/15.ui_extension/03.notifictions/notifications.md]: Fixed

**Review by sunilarjun — APPROVED** [review]
Thank you for addressing the changes!

**Pull request merged**